### ISSUE TITLE
S3 image gets blocked by CORS policy

### ISSUE THREAD

**Issue by DaniyalBasit** [Foliotek/Croppie#537]
Can you please add some more documentation on how to work with S3 images?
I have tried crossOrigin="anonymous" on the image and I have also tried to save the image in local storage and then use it, but it's not working me at all.

**Comment by diegoaa** [comment]
I can confirm. Reusing a jsfiddle someone has created for another issue: https://jsfiddle.net/w2L8o8er/ This displays the image in Firefox, but if I change the url from http://foliotek.github.io/Croppie/demo/demo-1.jpg to an S3 public image url (eg https://s3.amazonaws.com/github.issue/image.jpg ) it does not. However, the network monitor shows that the image is being fetched from S3.

**Comment by DaniyalBasit** [comment]
Thanks, @diegoaa, for explaining it further

**Comment by diegoaa** [comment]
I got it to work by following the instructions in https://stackoverflow.com/questions/38365182/image-saved-con-amazon-s3-bucket-not-showing-on-html5-canvas-crossorigin-issue. Changed the CORS configuration on my bucket, specifically the line
`<AllowedHeader>Authorization</AllowedHeader>`
to
`<AllowedHeader>*</AllowedHeader>`

**Comment by DaniyalBasit** [comment]
@diegoaa Thanks for sharing. I have followed this before. 
http://s3-ap-south-1.amazonaws.com/we-roll/products/145/original/RackMultipart20180611-29302-1nmndhj.jpg?1528729122
I am using the above URL, can you tell what wrong with it?. Is it because of the bucket name that is missing at the beginning of the URL.

**Comment by diegoaa** [comment]
I suspect your CORS configuration is not proper. https://www.test-cors.org gives me
`Fired XHR event: error` for that URL while my URL (now) gives me 200. Try with a different/fresh bucket, or paste your CORS configuration here?

**Comment by DaniyalBasit** [comment]
Bucket Policy:
{
    "Statement": [
        {
            "Sid": "AllowPublicRead",
            "Effect": "Allow",
            "Principal": {
                "AWS": "*"
            },
            "Action": "s3:GetObject",
            "Resource": "arn:aws:s3:::we-rolls/*"
        }
    ]
}

CORS Configuration:
<?xml version="1.0" encoding="UTF-8"?>
<CORSConfiguration xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
<CORSRule>
    <AllowedOrigin>*</AllowedOrigin>
    <AllowedMethod>GET</AllowedMethod>
    <MaxAgeSeconds>3000</MaxAgeSeconds>
    <AllowedHeader>*</AllowedHeader>
</CORSRule>
</CORSConfiguration>

**Comment by diegoaa** [comment]
Your CORS Configuration is the same as mine, though my bucket policy is blank. So if your bucket policy is not critical, try deleting it and instead setting object-level public read access, see if that works.

**Comment by DaniyalBasit** [comment]
I have done what you recommended, but still no luck.
Can you share an image link from your bucket?

**Comment by diegoaa** [comment]
https://s3.amazonaws.com/github.issue/image.jpg I'll keep this up for a few days for you to test out. All the best!
